### DISCUSSION TITLE
questions about units convert

### DISCUSSION THREAD

**Discussion by yuchanpei** [mir-group/nequip#249] [Q&A] [answered]
Dear developers,
I am trying to train a nequip potential with data in extxyz format. I use the following code to construct the dataset in jupyter notebook：
`E = float(energy_dict[pdbid])`
`sdfmol = ase.Atoms(symbols_str,positions=xyz_tuple_list)`
`calculator = SinglePointCalculator(sdfmol, energy=E)`
 `sdfmol.calc = calculator`
`io.write("test_ligand_pocket_with_spcalc.xyz",sdfmol, format='extxyz', append=True`
The unit of energy I use is kcal/mol. Should I convert the unit from kcal/mol to eV when I construct my own dataset?
When I use kcal/mol as an energy unit to train a nequip, should I use energy_units_to_eV in NequIPCalculator.from_deployed_model() to convert the energy unit since I use the ASE dataset loader to load my data?

Thank you in advance for any help you might be able to provide.

Best regards,
Cristina

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi Cristina,

Thanks for your interest in our code!

In general `nequip` is agnostic to units; the trained model will predict in whatever units the data is in. (Of course, the data must be self-constant and all in the same units.)

A **very important** caveat to this is `dataset: ase`, which inherits the unit conversion behavior of `ase.io.read`. This can be a bit subtle: if your data is in kcal/mol, for example, but in an `extxyz` file, ASE can't know that and will assume it is eV (ASE units) and pass it along to `nequip`, yielding a `nequip` model with kcal/mol units. However for some formats (mostly those coming out of DFT codes), ASE "knows" the units of the data and will convert them to eV and Angstrom in `ase.io.read`. Thus you will get a `nequip` model in eV and Angstrom. You can easily check what ASE is doing with your files by playing with `ase.io.read` in a Python REPL.

For running simulations in LAMMPS, it is important to use the correct LAMMPS units to match those predicted by the network (which are those of the data)--- quoting [the tutorial](https://colab.research.google.com/drive/1_r348f6oIyKxH4FnpKeD8g4QjwDhP8mT#scrollTo=JQs0ijPhvAGb&line=3&uniqifier=1):
> CAUTION: the reference data here are in kcal/mol for the energies and kcal/mol/A for the forces. The NequIP model will therefore also be predicting outputs in these units. We are therefore using `units real` in LAMMPS (see [docs](https://docs.lammps.org/units.html)). If your reference data are in other units, you should using the corresponding units command in LAMMPS (e.g. if you use eV, A then `units metal` would be appropriate, which would then also change time units from `fs` to `ps`).

For the ASE calculator, `NequIPCalculator`, we offer the `energy_units_to_eV` in case you trained a model that is **not** in eV, but that you want to use to run MD in ASE, which requires it to be in eV.  `NequIPCalculator` is not relevant for `dataset: ase` or training, just inference.
